Mathlib/AlgebraicGeometry/Morphisms/LocalClosure.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+
+import Mathlib.AlgebraicGeometry.Morphisms.Basic
+
+/-!
+# Local closure of morphism properties
+
+We define the source local closure of a property `P` wrt. a morphism property `W` and show it
+inherits stability properties from `P`.
+-/
+
+universe u
+
+open CategoryTheory Limits MorphismProperty
+
+namespace AlgebraicGeometry
+
+variable (W : MorphismProperty Scheme.{u})
+
+/-- The source (Zariski-)local closure of `P` is satisfied if there exists
+an open cover of the source on which `P` is satisfied. -/
+def SourceLocalClosure (P : MorphismProperty Scheme.{u}) : MorphismProperty Scheme.{u} :=
```

Comment:
I would prefer using the name `sourceLocalClosure` because the API favors thinking of morphism properties as terms in the type `MorphismProperty` rather as types themselves.